contracts/cw20-merkle-airdrop/src/state.rs

```diff
@@ -0,0 +1,23 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::Addr;
+use cw_storage_plus::{Item, Map, U8Key};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Config {
+    pub owner: Addr,
+    pub cw20_token_address: Addr,
+}
+
+pub const CONFIG_KEY: &str = "config";
+pub const CONFIG: Item<Config> = Item::new(CONFIG_KEY);
+
+pub const STAGE_KEY: &str = "stage";
+pub const STAGE: Item<u8> = Item::new(STAGE_KEY);
+
+pub const MERKLE_ROOT_PREFIX: &str = "merkle_root";
+pub const MERKLE_ROOT: Map<U8Key, String> = Map::new(MERKLE_ROOT_PREFIX);
```

Comment:
I wonder - as we want root would be always hex-parsed to `[u8; 32]`, why not to keep it in the map as `[u8; 32]`? It will both simplify implementation, and make contract more space efficient, so probably slightly cheaper.

Reply:
Actually not so with Map :disappointed: 

The content of Map are not stored raw, but rather JSON-encoded. This is usually a good choice.
The library we use based on serde-json-core will encode [u8; N] or Vec<u8> like any other Vec<T>, that is `[1, 5, 23, 175, 216, 82, ...]` Not very efficient.

We introduced the Binary wrapper to ensure base64 encoding/decoding.

So, Binary or (manually hex-encoded) String is probably better. Maybe Binary here

Reply:
That said, if you want performance and you are just going from `&[u8] -> [u8; 32]`, you could use raw storage get and set (and prefixed storage if you want that). That is definitely more efficient (no encoding going on), but I think makes the code more confusing